Sources/SwiftParser/ExpressionInterpretedAsVersionTuple.swift

```diff
@@ -0,0 +1,21 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(RawSyntax) import SwiftSyntax
+
+extension ExprSyntax {
+  public func asVersionTuple(with maxComponentCount: Int) -> VersionTupleSyntax? {
+    var parser = Parser(self.description)
+    let raw = parser.parseVersionTuple(maxComponentCount: maxComponentCount)
```

Comment:
You should make sure that all tokens in the `ExprSyntax` are represented idk the `VersionTupleSyntax`, similar to what we do when creating syntax nodes from string literals 

https://github.com/apple/swift-syntax/blob/e42bece52df2496d60b1b2a762fee9ffde7fc205/Sources/SwiftParser/generated/LayoutNodes%2BParsable.swift#L34

And now that I think about it, I think the best thing to do, would be add a parser entry function for `VersionTupleSyntax` so that all that code gets auto-generated, including the following, which is necessary for memory safety

https://github.com/apple/swift-syntax/blob/e42bece52df2496d60b1b2a762fee9ffde7fc205/Sources/SwiftParser/generated/LayoutNodes%2BParsable.swift#L23-L32


---

Could you add a test case for `2.0 + 1`. `+ 1` should be unexpected nodes in that case.

Reply:
I think making this a parser entry function is a good idea! And we have to pass the `maxComponentCount` parameter to that function so I add a helper function below.